src/main/java/luke/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package luke;
+
+import java.time.LocalDate;
+
+public class Deadline extends Task {
+    private LocalDate by;
+
+    /**
+     * Creates a Deadline with the specified name and date.
+     *
+     * @param name Name of task.
+     * @param by LocalDate representing when the deadline is by.
+     */
+    public Deadline(String name, LocalDate by) {
+        super(name);
+        this.by = by;
+    }
+
+    /**
+     * @return The status of task. ([taskType][isDone] taskname by)
+     */
+    @Override
+    public String fullStatus() { //TODO: add type of task later
```

Comment:
I think using a verb as part of the function name is more appropriate. E.g. getFullStatus ?